### PR TITLE
feat(benchpress): create component_benchmark macro

### PR DESCRIPTION
**Pull request by wagnermaciel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Currently we have a lot of repeat code for writing benchmark tests.

## What is the new behavior?
Reduce the amount of duplicate code for these tests by wrapping the boilerplate code in a macro.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by wagnermaciel** [comment]
### Questions
Is dev-infra the correct place to put this under in `.pullapprove.yml`?
Is this code located in the correct place?

**Review by jelbourn — APPROVED** [review]
LGTM

**Review by mhevery — CHANGES_REQUESTED** [review]
- [x] either squash or use `fixup!` https://dev.to/koffeinfrei/the-git-fixup-workflow-386d. This is a single unit of work and should have single commit when merged.
- [x] The `tools/components/BUILD.bazel` file is empty. Is that intentional? If so leave a comment as to why an empty file is needed.
- [x] Can we have a clear description in the commit message as to the motivation of this change.
- [x] I don't see any tests (or at least an example of working benchmark) How do I try it/know that it works.

**Comment by wagnermaciel** [comment]
@mhevery Done

**Review by mhevery — APPROVED** [review]

**Comment by wagnermaciel** [comment]
The only TAP failures are already-failing targets

**Comment by matsko** [comment]
presubmit = http://test/OCL:299857192:BASE:299857197:1583770689304:3497e3f2

**Comment by AndrewKushnir** [comment]
Hi @wagnermaciel, it looks like this PR has a conflict with master branch. Could you please rebase it and resolve the conflict (and add the `merge` label back)? Thank you.

File: tools/components/component_benchmark.bzl [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    """
+    Copies a file from tools/components/defaults to the destination.
+
+    Args:
+        origin: The name of a file in benchpress/defaults to be copied.
+        destination: Where the original file will be clopied to.
+    """
+    native.genrule(
+        name = "copy_default_" + origin + "_file_genrule",
+        srcs = ["//tools/components/defaults:" + origin],
```
Comment (devversion) [inline tools/components/component_benchmark.bzl]: Nit (for follow-up): It's more idiomatic to use substitutions. e.g. 

```python
"//tools/components/defaults:%s" % origin
```

Same could be done for all string concatenations in the file.

**Review by devversion — APPROVED** [review]
Just came across this. Awesome! LGTM

**Comment by AndrewKushnir** [comment]
@wagnermaciel, FYI given that we've just released `9.1.0-rc.0` and created a new patch branch (`9.1.x`), I've re-labeled this PR to be "master & patch", so that the changes are included into `9.1.0` release. Thank you.

**Pull request closed without merging**